tests/unit/test_vcs.py

```diff
@@ -299,6 +299,25 @@ def test_git_is_commit_id_equal(mock_get_revision, rev_name, result):
     assert Git.is_commit_id_equal("/path", rev_name) is result
 
 
+@pytest.mark.parametrize(
+    "version_out,result",
+    (
+        ("git version -2.25.1", ()),
+        ("git version 2.a.1", ()),
+        ("git ver. 2.25.1", ()),
+    ),
+)
```

Comment:
```suggestion
@pytest.mark.parametrize(
    "version_out, expected_message",
    (
        ("git version -2.25.1", "Unable to parse 'git version -2.25.1'"),
        ("git version 2.a.1", "Unable to parse 'git version 2.a.1'"),
        ("git ver. 2.25.1", "Unable to parse 'git ver. 2.25.1'"),
    ),
)
```

I would do this to avoid needing to format the log messasge in the test. When testing, repetition is sometimes better for explicitness.

Also I would use `record_tuple` instead to check the log is against WARNING.

Reply:
Thanks for the advice. Also, is there another way besides just importing logging and checking it against logging.WARNING?

Reply:
Each item in `record_tuple` is a three tuple `(level, logger_name, message)` so the canonical way is to `import logging`, and do

```python
assert caplog.record_tuple == [(logging.WARNING, "pip._internal.idontrememberthefullname", expected_message)]
```